test/util/oauthUtil.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable complexity, max-statements */
+define(function(require) {
+  var util = require('../util/util');
+  var OktaAuth = require('OktaAuth');
+  var tokens = require('./tokens');
+  var packageJson = require('../../package.json');
+
+  var oauthUtil = {};
+
+  // These are the result of the state and nonce after mocking
+  oauthUtil.mockedState = 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa';
+  oauthUtil.mockedNonce = oauthUtil.mockedState;
+
+  oauthUtil.mockStateAndNonce = function() {
+    // Make sure the state is generated the same every time (standardState, standardNonce)
+    spyOn(Math, 'random').and.callFake(function() {
+      return 0;
+    });  
+  };
+
+  var defaultPostMessage = {
+    'id_token': tokens.standardIdToken,
+    state: oauthUtil.mockedState
+  };
+
+  var defaultResponse = {
+    idToken: tokens.standardIdToken,
+    claims: tokens.standardIdTokenClaims
+  };
+
+  oauthUtil.setup = function(opts) {
+
+    if (opts &&
+        (opts.authorizeArgs && opts.authorizeArgs.responseMode !== 'fragment') ||
+        opts.getWithoutPromptArgs ||
+        opts.refreshArgs) {
+      // Simulate the postMessage between the window and the popup or iframe
+      spyOn(window, 'addEventListener').and.callFake(function(eventName, fn) {
+        if (eventName === 'message' && !opts.closePopup) {
+          // Call postMessage on the next tick
+          setTimeout(function() {
+            fn({
+              data: opts.postMessageResp || defaultPostMessage,
+              origin: opts.oktaAuthArgs && opts.oktaAuthArgs.url ||
+                'https://lboyette.trexcloud.com'
+            });
+          });
+        }
+      });
+    }
+
+    oauthUtil.mockStateAndNonce();
+
+    var authClient;
+    if (opts.oktaAuthArgs) {
+      authClient = new OktaAuth(opts.oktaAuthArgs);
+    } else {
+      authClient = new OktaAuth({
+        url: 'https://lboyette.trexcloud.com'
+      });
+    }
+
+    // Make sure our token isn't expired
+    var time;
+    if (opts.time || opts.time === 0) {
+      time = opts.time;
+    } else {
+      time = tokens.standardIdTokenClaims.exp - 1;
+    }
+    util.warpToUnixTime(time);
+
+    if (opts.hrefMock) {
+      util.mockWindowLocationHref(authClient, opts.hrefMock);
+    }
+
+    var promise;
+    if (opts.refreshArgs) {
+      promise = authClient.idToken.refresh(opts.refreshArgs);
+    } else if (opts.getWithoutPromptArgs) {
```

Comment:
I did have to add an optional getWithoutPromptArgs param to call getWithoutPrompt instead of authorize